src/Material/Options.elm

```diff
@@ -135,24 +175,44 @@ collect1' options acc =
     Class x -> { acc | classes = x :: acc.classes }
     CSS x -> { acc | css = x :: acc.css }
     Attribute x -> { acc | attrs = x :: acc.attrs }
+    Internal x -> { acc | internal = x :: acc.internal }
     Many options -> List.foldl collect1' acc options
     Set _ -> acc 
+    Listener event options decoder ->
+      { acc | listeners = (event, (decoder, options)) :: acc.listeners}
+    Lift m ->
+      { acc | dispatch = Just m}
     None -> acc
 
 
 collect' : List (Property c m) -> Summary () m 
 collect' = 
-  List.foldl collect1' (Summary [] [] [] ())
+  List.foldl collect1' (Summary [] [] [] [] [] Nothing ())
 
 
 addAttributes : Summary c m -> List (Attribute m) -> List (Attribute m)
-addAttributes summary attrs = 
-  List.append
-    attrs
-    (  Html.Attributes.style summary.css 
-    :: Html.Attributes.class (String.join " " summary.classes) 
-    :: summary.attrs
-    )
+addAttributes summary attrs =
+  let
+    listeners =
+      (case summary.dispatch of
+         Just lift ->
+           Dispatch.listeners lift (Dispatch.group summary.listeners)
+         Nothing ->
+           Dispatch.listeners' (Dispatch.group summary.listeners)
+      )
+  in
+    {- NOTE: Ordering here is important, First apply summary attributes
+    that way internal class and specific attributes can override those
+    provided by the user
+     -}
+    List.append
+      summary.attrs
+      (  Html.Attributes.style summary.css
+      :: Html.Attributes.class (String.join " " summary.classes)
+      :: attrs
+      )
+      |> (flip (++) summary.internal)
```

Comment:
The expression `A ++ B` is O(n) in the size of `A`, i.e., it traverses the `A` list, when it reaches the end, it adds `B`. The `List.append` is in principle the same as `++`, but in Elm, it turns out `(++)` is a [native implementation](https://github.com/elm-lang/core/blob/4.0.5/src/Native/Utils.js#L246-L270) and `List.append` is [implemented](https://github.com/elm-lang/core/blob/4.0.5/src/List.elm#L286-L295) in terms of `fold`.

It might make a difference if you used either `List.concat` or put `++` between each list here.